lnwallet/signer.go

```diff
@@ -0,0 +1,33 @@
+package lnwallet
+
+import (
+	"fmt"
+
+	"github.com/roasbeef/btcd/btcec"
+	"github.com/roasbeef/btcd/chaincfg/chainhash"
+)
+
+// MessageSigner is used for creation the signatures using the node identity key.
```

Comment:
Referring back to the initial specification, the `MessageSigner` should be an _interface_`. It should be a an interface with a single message as such: 
```go
type MessageSigner interface {
   SignMessage([]byte) (*btcec.Signature, error) 
}
```

With that, the wallet itself (via it's method or some dependent struct), _implement_  a single instance of the `MessageSigner` interface. The wallet shouldn't know of any distinction between it's funding keys and its identity keys. The wallet's only duty is to sign messages with the keys it's aware of. 

The constructor below violates the abstractions over private keys we maintain within the project (ignoring the lingering identity private key in the server which will be remedied at a later date). It doesn't belong in the wallet as the wallet isn't concerned with any identity management on the p2p layer, only the handling of Bitcoin specific keys.